example/pubspec.lock

```diff
@@ -505,14 +505,14 @@ packages:
       name: web_socket_channel
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.1.0"
+    version: "2.2.0"
```

Comment:
Nit: Maybe separate the updated pubspec into another commit next time, since it isn't really necessary for this change. But this is really a minor issue.

Reply:
agree

To be honest, I think `pubspec.lock` should be added to `gitignore` since it is a generated pub file. 

Yes, the dart documentation also [mentions this](https://dart.dev/guides/libraries/private-files#the-rules). So we can remove it from the example project but must leave it in packages.

Reply:
```
pubspec.lock  # Except for application packages
```
I interpret this to mean that it should be .gitignore'd for libraries but not for applications. Technically, the example is an application, not a library.

Reply:
ah, yeah, you're right 😂